### ISSUE TITLE
Neural network cache lookup optimization

### ISSUE THREAD

**Issue by shindavid** [shindavid/AlphaZeroArcade#40]
Right now, the softmax temperature is part of the neural network cache key. This was to avoid the cost of either performing a  softmax operation on cache-retrieved evaluations or storing a softmax'ed policy vector as a `Node` member (since the temperature-adjustment must be applied prior to the softmax operation).

Once we implement https://github.com/shindavid/blokus/issues/35, however, this decision will no longer makes sense. We should stored the softmax'ed policy vector a `Node` member instead.
